tests/test_fixture.py

```diff
@@ -13,6 +15,20 @@ def test_db_has_data(test_session):
         ['MoTIe', 'EC_raw', 'FLNRO-WMB', 'ENV-AQN'])
 
 
+def test_matview_path_exception(test_path_exception):
+    with pytest.raises(exc.ProgrammingError) as e:
+        brandy_hist = History(station_name='Brandy')
+        moti = Network(name='MoTIe')
+        station = Station(native_id='11091', network=moti,
+                          histories=[brandy_hist])
+
+        test_path_exception.query(Station.native_id)
+        test_path_exception.add(station)
```

Comment:
I wasn't sure how to implement process here, but tested a query which is what I believe is used in `align()` and insert which is used in `insert()` right now this raises a permission denied rather than `sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedFunction)`

Reply:
Yeah, so you'll probably want to `assert` something about the text of the Exception that gets raised. E.g. `assert "matview_queue_refresh_row" in e.message` or something like that. That will help you differentiate between the error that we're *trying* to test for, and other errors that shouldn't happen.

Reply:
You also want to isolate the `with` block to as narrow a window of code as can possibly raise the error (which should be inserting an observation `Obs`).